src/pytorch_lightning/accelerators/hpu.py

```diff
@@ -52,13 +55,28 @@ def get_parallel_devices(devices: int) -> List[torch.device]:
 
     @staticmethod
     def auto_device_count() -> int:
-        """Get the devices when set to auto."""
-        # TODO(@kaushikb11): Update this when api is exposed by the Habana team
-        return 8
+        """Returns the number of HPU devices when the devices is set to auto."""
+        try:
+            return torch_hpu.device_count()
+        except AttributeError:
+            rank_zero_debug("HPU auto_device_count failed, returning default count of 8.")
+            return 8
 
     @staticmethod
     def is_available() -> bool:
-        return _HPU_AVAILABLE
+        """Returns a bool indicating if HPU is currently available."""
+        try:
+            return torch_hpu.is_available()
```

Comment:
What's the difference between `torch_hpu.is_available` and `_HPU_AVAILABLE = habana_frameworks.torch.utils.library_loader.is_habana_avaialble()`?

Should this still use `_HPU_AVAILABLE` and instead update it's definition in `imports.py`?

Reply:
_HPU_AVAILABLE checks for the availability of the packages and drivers ensuring device can be uses and is_habana_available() is post creation of a synapse session to make queries to the library like device_count, name etc